core/src/impl/Kokkos_SharedAlloc.hpp

```diff
@@ -54,7 +54,15 @@
 #define KOKKOS_IMPL_IF_ON_HOST if (!__builtin_is_device_code())
 #else
 // Note: OpenMPTarget enforces C++17 at configure time
-#define KOKKOS_IMPL_IF_ON_HOST if constexpr (omp_is_initial_device())
+#pragma omp begin declare variant match(device = {kind(host)})
+static constexpr bool omp_is_initial_device2() { return true; }
```

Comment:
lets call this differently, to make clear it is NOT a OpenMP runtime function. How about kokkos_omp_on_device()